File: python/token_tls.py
import os
import pulsar

if __name__ == '__main__':
    service_url = os.environ['PULSAR_SERVICE_URL']
    topic = os.environ['PULSAR_TOPIC']
    token = os.environ['PULSAR_TOKEN']

    auth = pulsar.AuthenticationToken(token)
    client = pulsar.Client(service_url, authentication=auth)
    producer = client.create_producer(topic)
    msg_id = producer.send('hello'.encode())
    print(f'Sent to {msg_id}')
    producer.close()
    client.close()

File: node/token_tls.js
const Pulsar = require('pulsar-client');

(async () => {
    Pulsar.Client.setLogHandler((level, file, line, message) => {
        console.log('[%s][%s:%d] %s', Pulsar.LogLevel.toString(level), file, line, message);
    });

    const auth = new Pulsar.AuthenticationToken({
        token: process.env.PULSAR_TOKEN,
    })

    const client = new Pulsar.Client({
        serviceUrl: process.env.PULSAR_SERVICE_URL,
        authentication: auth,
    });

    const producer = await client.createProducer({
        topic: process.env.PULSAR_TOPIC,
    });

    const msg_id = await producer.send({
        data: Buffer.from(`hello`),
    });
    console.log(msg_id)

    await producer.close();
    await client.close();
})();
